test/performance/shmem_perf_suite/target_put.h

```diff
@@ -107,34 +121,29 @@ static inline void target_data_uni_bw(int len, perf_metrics_t metric_info)
     } else if (snode) {
         for (i = 0; i < num_partners; i++) {
             for(j = 0; j < metric_info.trials; j++) {
+                for(k = 0; k < metric_info.window_size; k++) {
 #ifdef USE_NONBLOCKING_API
-                shmem_putmem_nbi(metric_info.dest, metric_info.src, len, my_PE_partners[i]);
+                    shmem_putmem_nbi(metric_info.dest, metric_info.src, len, my_PE_partners[i]);
 #else
-                shmem_putmem(metric_info.dest, metric_info.src, len, my_PE_partners[i]);
+                    shmem_putmem(metric_info.dest, metric_info.src, len, my_PE_partners[i]);
 #endif
+                }
+                shmem_quiet();
             }
-            shmem_quiet();
             shmem_int_atomic_inc(&completion_signal, my_PE_partners[i]);
         }
     }
 
-    shmem_barrier_all();
+    //shmem_barrier_all();
```

Comment:
why is this commented out?

Reply:
Yes, I was going to ask this. I am not sure why we need a barrier at this point. Both sender and target are synchronized through the AMO and wait-until, right?